easybuild/tools/downloadtools.py

```diff
@@ -0,0 +1,187 @@
+            _log.debug('response code for given url %s: %s' % (url, url_fd.getcode()))
+            write_file(path, url_fd.read())
+            _log.info("Downloaded file %s from url %s to %s" % (filename, url, path))
+            downloaded = True
+            url_fd.close()
+        except urllib2.HTTPError as err:
+            if 400 <= err.code <= 499:
+                _log.warning("url %s was not found (HTTP response code %s), not trying again" % (url, err.code))
+                break
+            else:
+                _log.warning("HTTPError occured while trying to download %s to %s: %s" % (url, path, err))
+                attempt_cnt += 1
+        except IOError as err:
+            _log.warning("IOError occurred while trying to download %s to %s: %s" % (url, path, err))
+            attempt_cnt += 1
+        except Exception, err:
+            _log.error("Unexpected error occurred when trying to download %s to %s: %s" % (url, path, err))
+
+        if not downloaded and attempt_cnt < max_attempts:
+            _log.info("Attempt %d of downloading %s to %s failed, trying again..." % (attempt_cnt, url, path))
+
+    if downloaded:
+        _log.info("Successful download of file %s from url %s to path %s" % (filename, url, path))
+        return path
+    else:
+        _log.warning("Download of %s to %s failed, done trying" % (url, path))
+        return None
+
+
+def download_svn(revision, url, path, timeout):
+    """
+    Download a svn repository and create a tarball out of it in the given path
+    timeout is ignored here
+    """
+    # transform url into something SvnRepository understands
+    if url.startswith('svn+'):
+        url = url[4:]
+    # revision could already be a part of the url, we don't want that
+    if url.endswith(str(revision)):
+        url = url[:-len(revision)]
+    # same for path
+    if path.endswith(str(revision)):
+        path = path[:-len(revision)]
+    path = os.path.join(path, 'repo')
+    # svn revisions are digit's so only get these, some packages have version='r1234'
+    int_revision = ''.join([x for x in revision if x.isdigit()])
+
+    try:
+        svnrepo = SvnRepository(url)
+    except AttributeError, err:
+        _log.debug('error checkout out svn repo %s', err)
+        raise err
+    svnrepo.export(path, int_revision)
+    # make a tarfile in the directory next to the repo
+    tarpath = os.path.join(path, '..', revision)
+    make_tarfile(tarpath, path)
+    # path will always have '/repo' in it, so the rm should be file
+    shutil.rmtree(os.path.join(path))
+    return tarpath
+
+
+def download_local_file(filename, url, path, timeout):
+    """
+    Dowloading a local file is just copying it
+    timeout is ignored here.
+    """
+    if not url.endswith(filename):
+        url = os.path.join(filename)
+    if url.startswith('file://'):
+        url = url[7:]
+    _log.debug("copying %s to %s", url, path)
+    try:
+        shutil.copy2(url, path)
+    except IOError, error:
```

Comment:
nitpicking: use `err` rather than `error` (just to keep it consistent with 99% of the others `except`s in the codebase